src/Command/ColdStorageFlagCommand.php

```diff
@@ -0,0 +1,155 @@
+     *
+     * @throws \Exception If cannot find dataset with provided UDI.
+     * @throws \Exception If cannot find dataset submission in dataset.
+     * @throws \Exception If stubfile not readable/accessible.
+     * @throws \Exception If infofile not readable/accessible.
+     *
+     * @return void
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $systemPerson = $this->entityManager->find(Person::class, 0);
+
+        $infoFileName = $input->getArgument('infofile');
+        if (file_exists($infoFileName)) {
+            $infoFileContents = file($infoFileName);
+            $udi = trim($infoFileContents[0]);
+
+            $nudi = preg_replace('/:/', '.', $udi);
+            $infoPath = pathinfo($infoFileName)['dirname'];
+            $size = preg_replace('/size \(bytes\): /', '', trim($infoFileContents[1]));
+            $hash = preg_replace('/orig sha256: /', '', trim($infoFileContents[2]));
+            $originalFilename = preg_replace('/original filename: /', '', trim($infoFileContents[3]));
+            $stubFileName = "$infoPath/$nudi-manifest.zip";
+
+            if (file_exists($stubFileName)) {
+                $output->writeln("UDI: ($udi)");
+                $output->writeln("Original Size: ($size)");
+                $output->writeln("Original Hash: ($hash)");
+                $output->writeln("Original Filename: ($originalFilename)");
+                $output->writeln("stubFileName: ($stubFileName)");
+                $output->writeln("Attempting to flag $udi as Cold Stored.");
+
+                $datasets = $this->entityManager->getRepository(Dataset::class)->findBy(array('udi' => $udi));
+
+                if (count($datasets) == 0) {
+                    throw new \Exception('Could not find a dataset with the udi provided.');
+                } else {
+                    $output->writeln('Dataset Found.');
+                }
+                $dataset = $datasets[0];
+
+                $datasetSubmission = $dataset->getDatasetSubmission();
+                if (!($datasetSubmission instanceof DatasetSubmission)) {
+                    throw new \Exception('Could not find Dataset Submission.');
+                } else {
+                    $output->writeln('Submission Found.');
+
+                    // Set Modifier
+                    $datasetSubmission->setModifier($systemPerson);
+
+                    // Set filesize, hash, and original filename. (Cold Storage attributes).
+                    $datasetSubmission->setDatasetFileColdStorageAttributes($size, $hash, $originalFilename);
+
+                    // Set options for a new replacement datafile of the supplied Cold-Storage stubfile.
+                    $datasetSubmission->setDatasetFileTransferStatus(DatasetSubmission::TRANSFER_STATUS_NONE);
+                    $datasetSubmission->setDatasetFileName(null);
+                    $datasetSubmission->setDatasetFileSize(null);
```

Comment:
I just noticed that before filer runs, the dataland template breaks because of a null here. I don't think we should fix on this ticket, and the bug exists in production, so I'll put it in the backlog. If for whatever reason we run this and filer isn't running, this will be a temporary issue.